### PR TITLE
chore: #11633 is reverted manually

### LINKED ISSUE #12771: Clean up MME_BENCHMARK
MME_BENCHMARK is a new flag that also includes files via pre-processor mechanics.
* See https://github.com/magma/magma/pull/11633 for the introduction
* ~~Figure out if this flag is used in any CI builds~~
  * if not: ~~ask in slack if it can be removed and remove~~
  * ~~if yes: support the flag and by that bazelify missing files~~
* If this flag/build is supported, then also the following files should get bazelified
```
"./lte/gateway/c/core/oai/tasks/mme_app/experimental/mme_app_serialization.hpp"
"./lte/gateway/c/core/oai/tasks/mme_app/experimental/mme_app_serialization.cpp"
```
* Update bazel/scripts/check_c_cpp_bazel.sh

Update: See https://magmacore.slack.com/archives/C01B32Q9L76/p1662119897091189?thread_ts=1661851635.871369&cid=C01B32Q9L76
Can be removed. Make target also needs to be removed.

### PR DESCRIPTION
**Pull request by nstng**
Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

## Summary

* mme benchmark feature was introduced in #11633 but is not used anymore
* a discussion about the status was initiated because of the question if this needs to be modeled in bazel
* opinion of the original author @lionelgo via Slack https://magmacore.slack.com/archives/C01B32Q9L76/p1661851635871369
  * -> a removal is OK

This change is a manual revert of #11633, as a `git revert` shows a lot of merge conflicts because of later changes on affected files.

## Test Plan

* CI
  * feature was not used in CI, i.e., it's more or less a check if the removal was done wrong

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by vktng — APPROVED** [review]
LGTM.

**Review by voisey — APPROVED** [review]

**Review by jheidbrink — APPROVED** [review]

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp [hunk 1]
```diff
@@ -219,11 +219,10 @@ status_code_e MmeNasStateManager::read_ue_state_from_db() {
     for (const auto& key : keys) {
       OAILOG_DEBUG(log_task, "Reading UE state from db for %s", key.c_str());
       oai::UeContext ue_proto = oai::UeContext();
+      auto* ue_context = (ue_mm_context_t*)(calloc(1, sizeof(ue_mm_context_t)));
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp]: Please use the reinterpret_cast here.

Comment (nstng) [reply lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp]: Hi @ajahl, what is the interpretation here? This change was an unrelated refactoring to the (reverted) MME_BENCHMARK feature?

Changed back to the `reinterpret_cast` state (i.e., file is removed from change set).

Comment (ajahl) [reply lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp]: > what is the interpretation here?

`Reinterpret_cast` is the equivalent in C++ to C standard cast that is used in line 222. It is recommended by the C++ standard. And in C++ there are more specific casts, for example, `const_cast`, `static_cast`, `dynamic_cast` and so on. You don’t know which cast the C++ compiler will use when you apply the C standard cast, which may lead to an undefined behaviour. And there are more differences. The C standard cast breaks constness. In C++ you must explicitly use the const_cast for that. And in contract to C standard cast, you get an understandable error output.

**Review by ajahl — APPROVED** [review]

**Comment by rsarwad** [comment]
> > Hi @nstng, I have one concern, whether can we use this flag for checking the performance testing. As we are migrating AGW to use protobuf structures. Let me post my query on above slack channel
> 
> Hi @rsarwad, you could always revert this commit for local testing - but if it's not used in CI then I recommend to do not keep in in the code.

Sure @nstng. In above slack channel my query is answered and concluded that compile time flag, MME_BENCHMARK is not needed.

**Pull request merged**